### PR TITLE
[DO NOT MERGE] Delete smart contracts package

### LINKED ISSUE #9: A developer can delete packages they've previously created
Packages view allows a package to be deleted in IDE.
Removing the folder from the configured storage location also has the same effect.

From the current low-fis:

image

### PR DESCRIPTION
**Pull request by Jakeeyturner**
Closes #9

Signed-off-by: Jakeeyturner <jaketurner25@live.com>

### CODE REVIEW

File: client/src/commands/commandsUtil.ts [hunk 1]
```diff
@@ -75,4 +77,24 @@ export class CommandsUtil {
         return vscode.window.showQuickPick(runtimeNames, quickPickOptions);
     }
 
+    static async showSmartContractPackagesQuickPickBox(prompt: string): Promise<string | undefined> {
+        const blockchainPackageExplorerProvider = myExtension.getBlockchainPackageExplorerProvider();
+
+        const projects: Array<PackageTreeItem> = await blockchainPackageExplorerProvider.getChildren();
```
Comment (caroline-church) [inline client/src/commands/commandsUtil.ts]: This should be calling your new function

File: client/src/commands/commandsUtil.ts [hunk 2]
```diff
@@ -75,4 +77,24 @@ export class CommandsUtil {
...
+
+        const quickPickOptions = {
+            ignoreFocusOut: false,
+            canPickMany: false,
+            placeHolder: prompt
+        };
+
+        const projectNames = [];
```
Comment (caroline-church) [inline client/src/commands/commandsUtil.ts]: This needs a type

File: client/src/commands/deleteSmartContractPackageCommand.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    let packageToDelete: string;
+
+    if (packageTreeItem) {
+        packageToDelete = packageTreeItem.name;
+    } else {
+        packageToDelete = await CommandsUtil.showSmartContractPackagesQuickPickBox('Choose the smart contract package that you want to delete');
+    }
+
+    const blockchainPackageExplorerProvider = myExtension.getBlockchainPackageExplorerProvider();
+    const packages: Array<PackageTreeItem> = await blockchainPackageExplorerProvider.getChildren();
```
Comment (caroline-church) [inline client/src/commands/deleteSmartContractPackageCommand.ts]: This shouldn't be calling this it should use the new function

File: client/src/explorer/BlockchainPackageExplorer.ts [hunk 4]
```diff
@@ -32,34 +33,15 @@ export class BlockchainPackageExplorerProvider implements BlockchainExplorerProv
...
-                    vscode.window.showErrorMessage('Issue creating smart contract package folder:' + this.packageDir);
-                    return;
-                }
-            } else {
-                console.log('Error reading smart contract package folder:', error.message);
-                vscode.window.showErrorMessage('Issue reading smart contract package folder:' + this.packageDir);
-                return;
-            }
-        }
         this.tree = await this.createPackageTree(this.packageArray as Array<PackageTreeItem>);
```
Comment (caroline-church) [inline client/src/explorer/BlockchainPackageExplorer.ts]: this.packageArray isn't an array of packageTreeItems

File: client/src/util/CommandUtil.ts [hunk 5]
```diff
@@ -13,12 +13,16 @@
...
+import * as vscode from 'vscode';
 import * as child_process from 'child_process';
 import stripAnsi = require('strip-ansi');
 
 import * as childProcessPromise from 'child-process-promise';
 import { ConsoleOutputAdapter } from '../logging/ConsoleOutputAdapter';
 import { OutputAdapter } from '../logging/OutputAdapter';
+import * as fs from 'fs-extra';
+import * as homeDir from 'home-dir';
+import { PackageTreeItem } from '../explorer/model/PackageTreeItem';
```
Comment (caroline-church) [inline client/src/util/CommandUtil.ts]: This isn't used, if it is and i'm blind you shouldn't be creating packageTreeItems in here it should do this in the packageProvider

File: client/test/commands/commandsUtil.test.ts [hunk 6]
```diff
@@ -126,4 +128,28 @@ describe('Commands Utility Function Tests', () => {
         });
     });
 
+    describe('getPackages', () => {
+        // Should move and rewrite aspects of 'getChildren' tests from blockchainPackageExplorer and move them here.
```
Comment (caroline-church) [inline client/test/commands/commandsUtil.test.ts]: do you still need this comment?

File: client/src/commands/deleteSmartContractPackageCommand.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+        if (packageDir.startsWith('~')) {
+            // Remove tilda and replace with home dir
+            packageDir = homeDir(packageDir.replace('~', ''));
+        }
+
+        await fs.remove(packageDir + '/' + packages[index].name);
+
+    }
+    return blockchainPackageExplorerProvider.refresh();
```
Comment (Jakeeyturner) [inline client/src/commands/deleteSmartContractPackageCommand.ts]: this should call the refresh command eg vscode.commands.execute('blockchainAPackageExplorer.refreshEntry')

File: client/src/commands/deleteSmartContractPackageCommand.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+    let packageToDelete: string;
+
+    if (packageTreeItem) {
+        packageToDelete = packageTreeItem.name;
+    } else {
+        packageToDelete = await UserInputUtil.showSmartContractPackagesQuickPickBox('Choose the smart contract package that you want to delete');
+    }
+
+    const packages: any[] = await CommandUtil.getPackages();
```
Comment (caroline-church) [inline client/src/commands/deleteSmartContractPackageCommand.ts]: is the type string here if so we should change it to be string

File: client/src/util/CommandUtil.ts [hunk 9]
```diff
@@ -69,4 +71,46 @@ export class CommandUtil {
             });
         });
     }
+
+    public static async getPackages(): Promise<string[]> {
+        console.log('CommandUtils: getPackages');
+        let packageArray: Array<any> = [];
```
Comment (caroline-church) [inline client/src/util/CommandUtil.ts]: this is an array of strings

**Pull request closed without merging**